Assets/_BForBoss/_Core/Scripts/Boss/DerekContextManager.cs

```diff
@@ -0,0 +1,175 @@
+        
+        private RingLaborManager debug_rlm;
+        private DerekBossManager debug_dbm;
+        private Phase debug_currentPhase = Phase.Tutorial;
+        private Vulnerability debug_currentVulnerability = Vulnerability.Invulnerable;
+        
+        private void Debug_Initialize()
+        {
+            debug_rlm = FindObjectOfType<RingLaborManager>();
+            debug_dbm = FindObjectOfType<DerekBossManager>();
+            
+            Initialize(debug_rlm, debug_dbm);
+        }
+
+        [Button]
+        private void Debug_Reset()
+        {
+            debug_currentPhase = Phase.Tutorial;
+            debug_currentVulnerability = Vulnerability.Invulnerable;
+        }
+
+        [Button]
+        private void Debug_AdvancePhase()
+        {
+            if (debug_rlm == null || debug_dbm == null)
+            {
+                Debug_Initialize();
+            }
+
+            debug_currentPhase++;
+            debug_dbm.UpdatePhase(debug_currentPhase);
+            
+            if (debug_currentPhase == Phase.Death)
+            {
+                Debug_Reset();
+            }
+        }
+
+        [Button]
+        private void Debug_SwitchVulnerability()
+        {
+            if (debug_rlm == null || debug_dbm == null)
+            {
+                Debug_Initialize();
+            }
+
+            switch (debug_currentVulnerability)
+            {
+                case Vulnerability.Invulnerable:
+                    debug_currentVulnerability = Vulnerability.Vulnerable;
+                    break;
+                case Vulnerability.Vulnerable:
+                    debug_currentVulnerability = Vulnerability.Invulnerable;
+                    break;
+            }
+            
+            debug_dbm.UpdateVulnerability(debug_currentVulnerability);
+        }
+#endif
+        
+        
+        public void Reset()
+        {
+            if (_ringLaborManager == null || _bossManager == null)
+            {
+                return;
+            }
+            
+            _ringLaborManager.Reset();
+            _bossManager.Reset();
+            
+            //TutorialEndButton.Bind(this, BeginEncounter); 
+        }
+
+        public void Initialize(RingLaborManager ringLaborManager, DerekBossManager bossManager)
+        {
+            _ringLaborManager = ringLaborManager;
+            _bossManager = bossManager;
+            
+            //TutorialEndButton.Bind(this, OnPhaseChanged); 
+            _LaborIncompleteWaitForSeconds = new WaitForSeconds(_LaborIncompleteTimer);
+        }
+        
+        private void OnLaborCompleted(bool wasSuccessful)
+        {
+            if (_currentPhase == Phase.Tutorial || _currentVulnerability != Vulnerability.Invulnerable)
+            {
+                Perigon.Utility.Logger.LogError("Labor should not have have been active during the tutorial phase, or when the boss is not invulnerable. Exiting Encounter");
+                return;
```

Comment:
So if there's an error, is the boss just going to be stuck? how should the game handle it?  

-  Could reset the game? 
- crash the game? 
- Load with default settings?